### PR TITLE
include topic names in shadow debug messages

### PR DESCRIPTION
**Pull request by rgitzel**
include topic names in debug messages

Description
-----------
I found that referring to topics by name in the logs helped me understand what was going on.

I've also added logs in a handful of places that I was stuck on.

Checklist:
----------
*QUESTION:* How do I do these?  I've tested manually, but haven't found how to run automated tests or linting.
- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

### CODE REVIEW

File: lib/include/aws_shadow.h [hunk 1]
```diff
@@ -294,7 +294,7 @@ typedef struct ShadowCallbackParams
     /**
      * @brief Called to notify users that a delta document has been generated.
      *
-     * Callback for a message on /delete/accepted. Set to NULL for no callback.
+     * Callback for a message on /update/delta. Set to NULL for no callback.
```
Comment (rgitzel) [inline lib/include/aws_shadow.h]: I assume this was a copy and paste that wasn't corrected.

**Comment by sarenameas** [comment]
Thank you rgitzel for the helpful improvement suggestions. Our team is still working on procedures for accepting external PRs.

**Comment by rgitzel** [comment]
@sarenameas that's.... disappointing.

Could you please put a strong notice to that effect in this repo?  At the very least in the boilerplate for PR comments.  

I spent a couple hours making this a nice clean PR submission, and now that's turned out to be a waste of my time.

**Comment by sarenameas** [comment]
@rgitzel Yes. I apologize I do not want you to feel you have wasted time. I have created an issue in our internal request system. I will create a public issue for this.

**Comment by sarenameas** [comment]
Hello again @rgitzel , in the ble_beta branch you can find that Shadow has the thing name. The topic name is too verbose to print, it could be potentially long (https://github.com/aws/amazon-freertos/issues/397). We would have to make sure the customer increases the logging buffer size in FreeRTOSConfig.h.

**Comment by rgitzel** [comment]
@sarenameas

For the logging buffer size `configLOGGING_MAX_MESSAGE_LENGTH`, the examples I've found set it to 128 characters.   Including the full topic name isn't close to that:
```[ShDemoUpdt] [Shadow 0] MQTT: Publish to operation topic '$aws/things/rodney_esp32/shadow/update' succeeded.```
That is only 109 characters.   We could take out `$aws/things` as redundant, and be even shorter with no loss of information.

Regardless, there is no danger of buffer overrun, as the logging method already truncates messages longer than `configLOGGING_MAX_MESSAGE_LENGTH`(https://github.com/aws/amazon-freertos/blob/master/demos/common/logging/aws_logging_task_dynamic_buffers.c#L179).  

I've found having the topic name explicitly in the logs to be critical for debugging, as you otherwise have to *assume* which topic is being published to or read from.

**Pull request closed without merging**